Source/HTTP/WinHttp/winhttp_http_task.cpp

```diff
@@ -495,6 +505,8 @@ void winhttp_http_task::callback_status_request_error(
         HRESULT hr = HCHttpCallRequestGetUrl(pRequestContext->m_call, &method, &url);
         if (SUCCEEDED(hr))
         {
+            win32_cs_autolock autoCriticalSection(&pRequestContext->m_lock);
```

Comment:
Should this be release prior to calling complete_task? Does that synchronously call back into client's code?

Reply:
It looks like it can, but XSAPI doesn't currently use this. I've updated the PR to not hold locks for complete_task.